subsys/bluetooth/host/l2cap.c

```diff
@@ -966,8 +966,15 @@ static void l2cap_chan_destroy(struct bt_l2cap_chan *chan)
 	/* Cancel ongoing work. Since the channel can be re-used after this
 	 * we need to sync to make sure that the kernel does not have it
 	 * in its queue anymore.
+	 *
+	 * In the case where we are in the context of executing the rtx_work
+	 * item, we don't sync as it will deadlock the workqueue.
 	 */
-	k_work_cancel_delayable_sync(&le_chan->rtx_work, &le_chan->rtx_sync);
+	if (k_current_get() != &k_sys_work_q.thread) {
```

Comment:
It looks like it's possible to get the work thread by inspecting the work item. Maybe it's better than hard coding the assumed work thread here?

```suggestion
	struct k_thread *rtx_work_thread =  le_chan->rtx_work->queue->thread;
	if (k_current_get() != rtx_work_thread) {
```

Reply:
it's how it's done in conn.c, but I can give a try yeah.